src/Curves/Spline.jl

```diff
@@ -0,0 +1,131 @@
+#=
+ MIT License
+
+ Copyright (c) 2010-present David A. Kopriva and other contributors: AUTHORS.md
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy  
+ of this software and associated documentation files (the "Software"), to deal  
+ in the Software without restriction, including without limitation the rights  
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell  
+ copies of the Software, and to permit persons to whom the Software is  
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all  
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR  
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,  
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE  
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER  
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  
+ SOFTWARE.
+ 
+ --- End License
+=#
+
+mutable struct Spline
+    N::Int
+    x::Array{Float64}
+    y::Array{Float64}
+    b::Array{Float64}
+    c::Array{Float64}
+    d::Array{Float64}
```

Comment:
```suggestion
    x::Vector{Float64}
    y::Vector{Float64}
    b::Vector{Float64}
    c::Vector{Float64}
    d::Vector{Float64}
```
`Array{Float64}` can be any multi-dimensional array, so the current version is not type stable.

Reply:
Isn't Vector an alias for Array{xxx,1}? Those are arrays, and not vectors.

Reply:
Yes, `Vector{Float64} == Array{Float64, 1}` but `!= Array{Float64}`, which is `Array{Float64, N} where {N}` (dimension as free parameter). You can also use that if you prefer it, doesn't matter. But please do not use `Array{Float64}` unless you really want arrays with possible different dimensions.